code/workspaces/auth-service/src/dataaccess/userRolesRepository.js

```diff
@@ -9,14 +9,37 @@ function UserRoles() {
 // In line with other repository functions, this returns a promise,
 // but the model document is first converted to an object so the spread operator works as expected.
 function getRoles(userId) {
-  return UserRoles().findOne({ userId }).exec().then(document => document.toObject());
+  return UserRoles().findOne({ userId }).exec()
+    .then((document) => {
+      try {
```

Comment:
I don't think you need to handle this by exception and could instead null check the document

```
if (!document) {
  document = await addEmptyRecordForNewUser(userId);
}

return document.toObject();
```

Reply:
Note that the .toObject() will fail if the add function doesn't return a Mongoose document so if we choose to do that there will need to be two return points.